### PR TITLE
Enable highlights in grids

### LINKED ISSUE #1683: Remake Digital Universe grids with grid class
The grids from digital universe are made with the RenderableDUMesh class. Since we now have a RenderableGrid class, they should be remade with that for consistency.

### PR DESCRIPTION
**Pull request by Roxeena**
Add the possibility to, for example, highlight every 5th line in the grid with a thicker line and/or a different color. 

The previous `RenderableDUMeshes` grids have now been converted to use the `RenderableGrid` class instead.
Closes #1683

### CODE REVIEW

File: data/assets/scene/digitaluniverse/grids.asset [hunk 1]
```diff
@@ -311,19 +306,20 @@ local plane10ly = {
     Rotation = {
       Type = "StaticRotation",
       Rotation = eclipticRotationMatrix
+    },
+    Scale = {
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:310]: Is this `StaticScale` really necessary?  If the grid is 1ly and the Size is 2*lightYear size a defined?

File: data/assets/scene/digitaluniverse/grids.asset [hunk 2]
```diff
@@ -338,19 +334,20 @@ local plane100ly = {
...
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:338]: Same here, I think defining the size to be `2 * 10 * lightYear` might be easier to understand

Comment (alexanderbock) [reply data/assets/scene/digitaluniverse/grids.asset:338]: Same for the ones below;  just to reduce the level of complexity by removing the need for specifying the `StaticScale` at all

File: data/assets/scene/digitaluniverse/grids.asset [hunk 3]
```diff
@@ -434,17 +438,22 @@ local plane100kly = {
...
     Opacity = 0.4,
-    File = speck .. "1Mly.speck",
-    MeshColor = {{ 0.1, 0.5, 0.6 }},
-    LabelFile = speck .. "1Mly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 19.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = {0.3, 0.7, 0.8 },
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:452]: Missing space after the `{`

File: data/assets/scene/digitaluniverse/grids.asset [hunk 4]
```diff
@@ -414,17 +413,22 @@ local plane10kly = {
...
-    LabelFile = speck .. "100kly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 18.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = { 0.3, 0.7, 0.8 },
+    LineWidth = 2.0,
+    Segments = { 20, 20 },
+    HighlightRate = { 5, 5},
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:430]: Missing space before the `}`

File: data/assets/scene/digitaluniverse/grids.asset [hunk 5]
```diff
@@ -434,17 +438,22 @@ local plane100kly = {
...
-    LabelFile = speck .. "1Mly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 19.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = {0.3, 0.7, 0.8 },
+    LineWidth = 2.0,
+    Segments = { 20, 20 },
+    HighlightRate = { 5, 5},
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:455]: Missing space before the `}`

File: data/assets/scene/digitaluniverse/grids.asset [hunk 6]
```diff
@@ -454,17 +463,22 @@ local plane1Mly = {
...
-    LabelFile = speck .. "10Mly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 20.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = { 0.3, 0.7, 0.8 },
+    LineWidth = 2.0,
+    Segments = { 20, 20 },
+    HighlightRate = { 5, 5},
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:480]: Missing space before the `}`

File: data/assets/scene/digitaluniverse/grids.asset [hunk 7]
```diff
@@ -474,17 +488,22 @@ local plane10Mly = {
...
-    LabelFile = speck .. "100Mly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 21.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = { 0.3, 0.7, 0.8 },
+    LineWidth = 2.0,
+    Segments = { 20, 20 },
+    HighlightRate = { 5, 5},
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:505]: Missing space before the `}`

File: data/assets/scene/digitaluniverse/grids.asset [hunk 8]
```diff
@@ -494,17 +513,22 @@ local plane100Mly = {
...
-    LabelFile = speck .. "20Gly.label",
-    TextColor = { 0.0, 0.2, 0.5 },
-    TextSize = 23.6,
-    TextMinMaxSize = { 0, 30 },
-    Unit = "Mpc"
+    Color = { 0.1, 0.5, 0.6 },
+    HighlightColor = { 0.3, 0.7, 0.8 },
+    LineWidth = 2.0,
+    Segments = { 40, 40 },
+    HighlightRate = { 5, 5},
```
Comment (alexanderbock) [inline data/assets/scene/digitaluniverse/grids.asset:530]: Missing space before the `}`

**Comment by alexanderbock** [comment]
Changing the "Highlight Rate" while OpenSpace is running doesn't update the highlighting

File: data/assets/scene/digitaluniverse/grids.asset [hunk 9]
```diff
@@ -434,17 +438,22 @@ local plane100kly = {
 
 local plane1Mly = {
   Identifier = "1MlyGrid",
+  Transform = {
+    Scale = {
+      Type = "StaticScale",
+      Scale = 1000000
```
[the remark this answers is not in the document]
Comment (WeirdRubberDuck) [reply data/assets/scene/digitaluniverse/grids.asset:444]: Or actually, saw that Alex already commented on this. And yeah, keeping the grid size to just one place might be a good idea

**Comment by Roxeena** [comment]
> Changing the "Highlight Rate" while OpenSpace is running doesn't update the highlighting

Hmm strange, it works for me. Did you change the color of the highlight so it isn't the same as the base color? And if the rate is larger than the grid it only highlights the center "cross" and then it might look like nothing happens.

File: modules/base/rendering/grids/renderablegrid.cpp [hunk 10]
```diff
@@ -42,19 +42,39 @@ namespace {
...
         "This value specifies the number of segments that are used to render the "
         "grid in each direction"
     };
 
+    constexpr openspace::properties::Property::PropertyInfo HighlightRateInfo = {
+        "HighlightRate",
+        "Highlight Rate",
+        "The rate that the columns and rows are highlighted, counted with respect to the "
+        "center of the grid. If the number of segments in the grid is odd, the "
+        "highlighting might be offsett from the center."
```
Comment (WeirdRubberDuck) [inline modules/base/rendering/grids/renderablegrid.cpp:63]: typo: "offset"

**Review by WeirdRubberDuck — APPROVED** [review]
Looks great :) 

Should we put an example of this in the examples/grids.asset or are the DU grids asset sufficient?

File: data/assets/scene/digitaluniverse/grids.asset [hunk 11]
```diff
@@ -311,19 +306,16 @@ local plane10ly = {
     Rotation = {
       Type = "StaticRotation",
       Rotation = eclipticRotationMatrix
-    }
+    },
```
Comment (WeirdRubberDuck) [inline data/assets/scene/digitaluniverse/grids.asset:309]: Remove these extra commas (everywhere)

**Pull request merged**